src/providers/hana/qgshananewconnection.cpp

```diff
@@ -0,0 +1,348 @@
+  if ( !settings.getMultitenant() )
+  {
+    rbtnSingleContainer->setChecked( true );
+    frmMultitenantSettings->setEnabled( false );
+  }
+  else
+  {
+    rbtnMultipleContainers->setChecked( true );
+    if ( settings.getDatabase() == QStringLiteral( "SYSTEMDB" ) )
+      rbtnSystemDatabase->setChecked( true );
+    else
+      txtTenantDatabaseName->setText( settings.getDatabase() );
+  }
+  txtSchema->setText( settings.getSchema() );
+  chkUserTablesOnly->setChecked( settings.getUserTablesOnly() );
+  chkAllowGeometrylessTables->setChecked( settings.getAllowGeometrylessTables() );
+
+  chkEnableSSL->setChecked( settings.getEnableSsl() );
+  int idx = cbxCryptoProvider->findData( settings.getSslCryptoProvider() );
+  if ( idx >= 0 )
+    cbxCryptoProvider->setCurrentIndex( idx );
+
+  chkValidateCertificate->setChecked( settings.getSslValidateCertificate() );
+  txtOverrideHostName->setText( settings.getSslHostNameInCertificate() );
+  txtKeyStore->setText( settings.getSslKeyStore() );
+  txtTrustStore->setText( settings.getSslTrustStore() );
+
+  if ( settings.getSaveUserName() )
+  {
+    mAuthSettings->setUsername( settings.getUserName() );
+    mAuthSettings->setStoreUsernameChecked( true );
+  }
+
+  if ( settings.getSavePassword() )
+  {
+    mAuthSettings->setPassword( settings.getPassword() );
+    mAuthSettings->setStorePasswordChecked( true );
+  }
+
+  mAuthSettings->setConfigId( settings.getAuthCfg() );
+
+  txtName->setText( settings.getName() );
+}
+
+void QgsHanaNewConnection::testConnection()
+{
+  QString warningMsg;
+  if ( txtDriver->text().isEmpty() )
+    warningMsg = QStringLiteral( "Driver name has not been specified." );
+  else if ( txtHost->text().isEmpty() )
+    warningMsg = QStringLiteral( "Host name has not been specified." );
+  else if ( rbtnMultipleContainers->isChecked() && rbtnTenantDatabase->isChecked() &&
+            txtTenantDatabaseName->text().isEmpty() )
+    warningMsg = QStringLiteral( "Database has not been specified." );
+  else if ( mAuthSettings->username().isEmpty() )
+    warningMsg = QStringLiteral( "User name has not been specified." );
+  else if ( mAuthSettings->password().isEmpty() )
+    warningMsg = QStringLiteral( "Password has not been specified." );
+  else if ( txtIdentifier->text().isEmpty() )
+    warningMsg = QStringLiteral( "Identifier has not been specified." );
+
+  if ( !warningMsg.isEmpty() )
+  {
+    bar->clearWidgets();
+    bar->pushWarning( tr( "Connection Failed" ), warningMsg );
+    return;
+  }
+
+  QgsTemporaryCursorOverride cursorOverride( Qt::WaitCursor );
+
+  QgsHanaSettings settings( txtName->text() );
+  readSettingsFromControls( settings );
+  QgsHanaConnectionRef connRef( settings.toDataSourceUri() );
+
+  if ( !connRef.isNull() )
+    bar->pushMessage( tr( "Connection to the server was successful." ), Qgis::Info );
+  else
+    bar->pushMessage( tr( "Connection failed - consult message log for details." ), Qgis::Warning );
```

Comment:
Can't we display the error message here?

Reply:
Done in a209501.